### PR TITLE
Remove ApmMiddleWare, only use DiagnosticSource listener for ASP.NET Core

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: src/integrations/Elastic.Apm.AspNetCore/ApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -80,11 +84,12 @@ public static class ApmMiddlewareExtension
...
 			var subs = subscribers?.ToList() ?? new List<IDiagnosticsSubscriber>(1);
 
-			if (subs.Count == 0 || subs.All(s => s.GetType() != typeof(AspNetCoreErrorDiagnosticsSubscriber)))
-				subs.Add(new AspNetCoreErrorDiagnosticsSubscriber());
+			if (subs.Count == 0 || subs.All(s => s.GetType() != typeof(AspNetCoreDiagnosticSubscriber)))
+				subs.Add(new AspNetCoreDiagnosticSubscriber());
 
 			agent.Subscribe(subs.ToArray());
-			return builder.UseMiddleware<ApmMiddleware>(agent.Tracer, agent);
+			//return builder.UseMiddleware<ApmMiddleware>(agent.Tracer, agent);
```
Comment (stevejgordon) [inline src/integrations/Elastic.Apm.AspNetCore/ApplicationBuilderExtensions.cs:91]: nit: can we remove this commented line

File: test/Elastic.Apm.Tests.Utilities/XUnit/MemberData.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@ public static IEnumerable<object[]> TestWithDiagnosticSourceOnly()
 		yield return new object[] { false };
 		//
 		// Skip "DiagnosticSourceOnly" tests on .NET 7
-		// until https://github.com/dotnet/aspnetcore/issues/45233 is resolved.
+		// until  is resolved.
```
Comment (stevejgordon) [inline test/Elastic.Apm.Tests.Utilities/XUnit/MemberData.cs:16]: Did we mean to remove the link here?

**Review by stevejgordon — DISMISSED** [review]
LGTM with a couple of nits

**Review by stevejgordon — APPROVED** [review]

File: sample/WebApiExample/Program.cs
Comment (cboudereau) [inline sample/WebApiExample/Program.cs:1]: Hi! I am really happy to see that you finally integrated one of my contribution of the webapi example : https://github.com/elastic/apm-agent-dotnet/pull/1643/files.

**Pull request merged**